### DISCUSSION TITLE
How to properly use ComputeCurveSampler

### DISCUSSION THREAD

**Discussion by NicusorN5** [stride3d/stride#1718] [Q&A] [answered]
I have this curve in the editor (set as Animation):
![image](https://github.com/stride3d/stride/assets/20599225/9cdb0f4a-e82c-40f0-ad15-6396838203e7)

All values are from `X ∈[0, 1]`, `Y ∈ [0, 1]`.

And I'm trying to test it like so:
```cs
            for (float i = 0; i < 1.0f; i += 0.05f)
            {
                Debug.Print(Curve.Evaluate(i) + "");
            }
```

All the outputed values are 0. Any help or ideas on this? Is this a bug, or am I misusing this feature?

Thanks.

**Comment by NicusorN5** [comment, marked as answer]
(Accidentally closed without posting)

I had to call `.UpdateChanges()` because a "baked array" stored in the instance was filled with 0 for some reason.

 Now it is showing results as expected. Solved.

**Reply by NicusorN5** [reply]
Funny how answering to my own question counts as a contribution 😆

**Reply by VaclavElias** [reply]
Probably we should keep it open for some time, also because it is Q&A? It might be helpful for others? GitHub defaults only to open discussions. A few clicks to access closed discussions :)

**Reply by NicusorN5** [reply]
Ok, reopened
